File: src/euler/level1/problem003.clj
(ns euler.level1.problem003
  (:use [euler.util.math :only [divisible-by? abs]]))

(defn euler-3 [n]
  (loop [num (abs n)
         div 2]
    (cond
      (>= div num)
        num
      (divisible-by? num div)
        (recur (/ num div) div)
      :else
        (recur num (inc div)))))

File: ruby/lib/problem003.rb
def divisible_by?(num, divisor)
  num % divisor == 0
end

def euler_3(n)
  n = n.abs
  return 1 if n <= 1

  divisor = 2
  while n != divisor
    if divisible_by?(n, divisor)
      n /= divisor
    else
      divisor += 1
    end
  end
  divisor
end
